.ci/set_dependency_version

```diff
@@ -0,0 +1,38 @@
+#! /usr/bin/env bash
+
+set -eu
+
+if [ -z "${MAIN_REPO_DIR:-}" ]; then 
+	MAIN_REPO_DIR="$(readlink -f "$(dirname "${0}")/..")"
+	export MAIN_REPO_DIR
+fi
+
+echo "repo-dir: $(MAIN_REPO_DIR)"
+
+if [ "${DEPENDENCY_NAME}" != "github.com/gardener/machine-controller-manager" ]; then 
+	echo "error: do not know how to upgrade ${DEPENDENCY_NAME}"
+	exit 1
+fi
+
+if [ -z "${DEPENDENCY_VERSION}" ]; then
+	echo "env var DEPENDENCY_VERSION must be set"
+	exit 1
+fi
+
+MCM_FILEPATH="${MAIN_REPO_DIR}/MCM_VERSION"
+
+if [ ! -f "${MCM_FILEPATH}" ]; then
+	echo "error no such file: ${MCM_FILEPATH}"
+	exit 1
+fi
+
+echo -n "${DEPENDENCY_VERSION}" > "${MCM_FILEPATH}"
+echo "set dependency-version of ${DEPENDENCY_NAME} to ${DEPENDENCY_VERSION}"
+
+cd ${MAIN_REPO_DIR}
+old_version=$(cat go.mod | grep "github.com/gardener/machine-controller-manager v" | xargs)
+new_version="github.com/gardener/machine-controller-manager${DEPENDENCY_VERSION}"
+
+sed -i -- 's#'"${old_version}"'#'"${new_version}"'#g' go.mod
+apk add --no-cache go make
+make revendor
```

Comment:
maybe breaking it into two steps(`go mod tidy` and `go mod vendor`) could be better, as I saw last auto-revendoring in other providers failed using the make rule but worked by using these two commands seperately.

Reply:
I think the issue was that we were calling vendor before tidy. I changed that and seem to fix the problem in my tests